DBZ-6240 Change stream supports filtering schema change event (#4385)
debezium/debezium: documentation/modules/ROOT/pages/transformations/schema-change-event-filter.adoc

The following excerpt from a connector configuration shows how you would add the transformation to the connector and set it to remove `DROP` and `TRUNCATE` events from the DDL messages that the connector captures.
[[example-configuration-schema-change-event-filter]]
.Schema change event filter configuration to remove `DROP` and `TRUNCATE` events

[source]
----
transforms=filterTableDropAndTruncateEvent
transforms.filterTableDropAndTruncateEvent.type=io.debezium.transforms.SchemaChangeEventFilter
transforms.filterTableDropAndTruncateEvent.schema.change.event.exclude.list=DROP,TRUNCATE
----

For more information about configuring the transformation, see the xref:schema-change-event-filter-configuration-options[schema change event filter configuration options].
[[schema-change-event-filter-effect-of-applying-the-schemachangeeventfilter-smt]]
== Effect of applying the `SchemaChangeEventFilter` SMT
After you configure the SMT for a connector, whenever the connector captures `DROP` or `TRUNCATE` events, it resets the values for these events to `null`.

> The connector does not send events with the resulting `null` values to Kafka.
> The following examples show how the transformation changes the values of event records that you designate for exclusion.
> .Schema change event record before it is processed by the `SchemaChangeEventFilter` transformation
> ====
> Schema change event record before it is processed by the `SchemaChangeEventFilter` transformation::
> 
> Schema change event record before the SMT processes the event record:::
> +

Reviewer: ```suggestion
The connector does not send events with the resulting `null` values to Kafka.

The following examples shows a xref:schema-change-event-filter-example-before-processing[schema change event record before it is processed] by the SMT, followed by the xref:schema-change-event-filter-example-after-processing[output after the SMT filters the event].

[[schema-change-event-filter-example-before-processing]]
.Schema change event record before it is processed by the `SchemaChangeEventFilter` transformation
====
```

                "optional":true,
                "field":"ddl"
            },
            {
                "type":"array",
                "items":{
                    "type":"struct",
                    "fields":[
                        {
                            "type":"string",
                            "optional":false,
                            "field":"type"
                        },
                        {
                            "type":"string",
                            "optional":false,
                            "field":"id"
                        }
                    ],
                    "optional":false,
                    "name":"io.debezium.connector.schema.Change"
                },
                "optional":false,
                "field":"tableChanges"
            }
        ],
        "optional":false,
        "name":"filter.SchemaChangeValue"
    },
    "payload":{
        "ts_ms":1691035505397,
        "databaseName":"test",
        "schemaName":"test_schema",
        "ddl":"",
        "tableChanges":[
            {
                "type":"DROP",
                "id":"test.table"
            }
        ]
    }
}
----

====

> .Schema change event record after processing by the SMT

Reviewer: ```suggestion

Based on the earlier xref:example-configuration-schema-change-event-filter[configuration example], after the SMT processes the xref:schema-change-event-filter-example-before-processing[original schema change event], it resets the values for the `DROP` event to `null`, as shown in the following example:

[[schema-change-event-filter-example-after-processing]]
.Schema change event record after processing by the SMT
```

====

[source]
----
null
----
====

> 
> 

Reviewer: ```suggestion

// Type: reference
```
